Zend/zend_weakrefs.c

```diff
@@ -14,11 +14,11 @@
    +----------------------------------------------------------------------+
 */
 
-#include "zend.h"
-#include "zend_interfaces.h"
-#include "zend_objects_API.h"
 #include "zend_weakrefs.h"
+#include "zend_API.h" // for ZEND_BEGIN_ARG_INFO_EX
+#include "zend_objects.h" // for zend_object_std_init()
 #include "zend_weakrefs_arginfo.h"
+#include "zend_interfaces.h" // for zend_create_internal_iterator_zval()
```

Comment:
```suggestion
#include "zend_weakrefs.h"
#include "zend_weakrefs_arginfo.h"
#include "zend_API.h" // for ZEND_BEGIN_ARG_INFO_EX
#include "zend_objects.h" // for zend_object_std_init()
#include "zend_interfaces.h" // for zend_create_internal_iterator_zval()
```
I think having the arginfo next to the corresponding file header is "better" except if there is a reason not to? As including an arginfo header makes it clear very early on that we are dealing with either a class or an extension defining functions.